reports/2017-05-14.md

```diff
@@ -0,0 +1,80 @@
+There is also an _experimental_ new Git packfile encoder and decoder called [Sirodepac](https://github.com/dinosaure/sirodepac) to let ocaml-git perform compression of repositories more easily, also by [@dinosaure].
+
+There has also been significant progress this week on having a filesystem that Irmin and DataKit can use when compiled as unikernels (and hence only have access to a raw block device):
+* [Tom Ridge](http://www.tom-ridge.com) [announced](https://lists.xenproject.org/archives/html/mirageos-devel/2017-05/msg00003.html) the [initial release](https://github.com/tomjridge/tjr_btree) of a formally verified btree filesystem. It still needs to be integrated with the MirageOS filesystem interfaces, so volunteers who want to glue things together are welcome to participate.
+* [Gabriel de Perthuis](https://github.com/g2p) also responded to note that he is continuing to work on a flash-optimised pure OCaml filesystem (using [hitchhiker trees](https://lists.xenproject.org/archives/html/mirageos-devel/2017-01/msg00002.html)). It is not quite ready to open-source yet, but he anticipates doing so in the next month or so.
+
+Between both of these efforts, it looks like a fully unikernel-aware, type-safe DataKit that persists onto a block storage device is not too far away.
+
+Finally, we welcome [David Udelson](http://www.davidudelson.com), a junior from Cornell who has been selected as a Google Summer of Code student to work on a REST API for Irmin!
+
+## Releases
+
+This week saw a minor [moby/datakit:0.10.1] release of DataKit to support the latest versions of the 9P and Lwt libraries.  Both of these have some backwards-imcompatible changes, so the 0.10.1 release of DataKit lets us use the latest features.
+
+## PRs merged
+
+- DataKit support for the latest 9P versions was added, including upstreaming the local changes to the 9P library that were only present in DataKit before. ([datakit#549] [datakit#550] [datakit#551] [datakit#552] [@talex5] [@djs55] [@samoht] [@avsm]).
+- Irmin no longer logs errors on invalid branch names. ([irmin#441] [irmin#440] [@samoht]).
+- Irmin support for Cstruct contents was fixed to pretty print the raw contents, not the hexdump. ([irmin#442] [@samoht]).
+- `Irmin.Hash.of_string` now does not raise an exception on invalid hash ([irmin#443] [@samoht]).
+- OCaml Git got a `commit-add-parent` command ([git#208] [@samoht]).
+- Git also switched to using the new Decompress library ([git#211] [git#145] [git#145] [git#210] [@dinosaure] [@samoht]).
+
+The Jbuilder build porting journey continues to be successful:
+
+- Irmin was ported to Jbuilder, speeding it up a lot. ([irmin#444] [@samoht]).
+- OCaml-Git was also ported to Jbuilder ([git#209] [@samoht]).
+- The build now also supports Mtime 1.0, which had some API incompatiblities from previous versions  ([git#212] [datakit#554] [@samoht]).
+- The `ogit` binary was temporarily missing, now fixed ([git#213] [@samoht]).
+
```

Comment:
The main dev effort this week was https://github.com/moby/datakit/pull/551 which made the client/server/tests independent of the transport used. The transport currently available is still 9p, but a Git client (without the server side) and gRPC transport are being working on.